packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -858,7 +834,22 @@ contract SphinxUtils is SphinxConstants, StdUtils {
 
     function getSphinxModuleAddress(SphinxConfig memory _config) public pure returns (address) {
         address safeProxyAddress = getGnosisSafeProxyAddress(_config);
-        bytes32 salt = keccak256(abi.encode(safeProxyAddress, safeProxyAddress, _config.saltNonce));
+        bytes32 salt = keccak256(
+            abi.encode(
+                safeProxyAddress,
+                safeProxyAddress,
+                // We always set the `saltNonce` of the Sphinx Module to `0` because the
+                // `sphinxConfig.saltNonce` field is only used when deploying the Gnosis Safe. It's
+                // not necessary to include the `saltNonce` here because a new Sphinx Module will be
+                // deployed if the user sets the `sphinxConfig.saltNonce` to a new value and then
+                // deploys a new Gnosis Safe using Sphinx's standard deployment method. A new Sphinx
+                // Module is deployed in this scenario because its address is determined by the
+                // address of the Gnosis Safe. It'd only be necessary to include a `saltNonce` for
+                // the Sphinx Module if a single Gnosis Safe wants to enable multiple Sphinx
+                // Modules, which isn't a feature that we currently support.
+                0
+            )
+        );
```

Comment:
This is the area that resolves the bug. We were previously using `_config.saltNonce` to calculate the Sphinx Module's address, which isn't correct because this field is only meant to determine the Gnosis Safe's address. Hard-coding this value to `0` resolves the bug